### PR TITLE
distribute end-user authn policy to mixerclient

### PR DESCRIPTION
**Pull request by ayj**
* define models for EndUserAuthenticationPolicySpec and
  EndUserAuthenticationPolicySpecBinding.

* update pilot to monitor and distribute end-user authn policy through
  mixerclient filter config.

* add envoy configuration test for mixerclient service configs

### CODE REVIEW

**Comment by ayj** [comment]
Trying again after manually deleted stale auth CRD from previous test runs.

/retest

File: pilot/test/mock/config.go [hunk 1]
```diff
@@ -118,6 +120,27 @@ var (
...
+				Audiences: []string{
+					"audience_foo.example.com",
+					"audience_bar.example.com",
+				},
+				JwksUri:                "https://www.example.com/oauth/v1/certs",
+				ForwardJwt:             true,
+				PublicKeyCacheDuration: types.DurationProto(5 * time.Minute),
+				Locations: []*mccpb.JWT_Location{{
+					Scheme: &mccpb.JWT_Location_Header{
+						Header: "Bearer",
```
Comment (qiwzhang) [inline pilot/test/mock/config.go]: Don't use "Bearer".  It is confusing.   "Bearer" is not a http header name.  It is a prefix in the http header value.

Comment (ayj) [reply pilot/test/mock/config.go]: I already fixed this elsewhere but missed this spot - thanks.

**Pull request merged**